src/features/metadata-based-view/MetadataBasedItemList.js

```diff
@@ -0,0 +1,148 @@
+import FileIcon from '../../icons/file-icon';
+import messages from '../../elements/common/messages';
+import { getFileExtension } from '../../utils/file';
+import '../../elements/content-explorer/ItemList.scss';
+import './MetadataBasedItemList.scss';
+
+const FILEICON_SIZE = 32;
+const FILEICON_COLUMN_WIDTH = 52;
+const FILENAME_COLUMN_WIDTH = 350;
+const MIN_METADATA_COLUMN_WIDTH = 250;
+const FIXED_COLUMNS_NUMBER = 2; // 2 Sticky columns - 1. file-icon, 2. file-name for each row
+const FIXED_ROW_NUMBER = 1; // Header row
+
+type Props = {
+    currentCollection: Object,
+    intl: Object,
+    metadataColumnsToShow: Array<string>,
+} & InjectIntlProvidedProps;
+
+type CellRendererArgs = {
+    columnIndex: number,
+    key: string,
+    rowIndex: number,
+    style: Object,
+};
+
+class MetadataBasedItemList extends Component<Props> {
+    props: Props;
+
+    getColumnWidth(width) {
+        const { metadataColumnsToShow }: Props = this.props;
+        const availableWidth = width - FILENAME_COLUMN_WIDTH - FILEICON_COLUMN_WIDTH; // total width minus width of sticky columns
+        // Maintain min column width, else occupy the rest of the space equally
+        const mdColumnWidth = Math.max(availableWidth / metadataColumnsToShow.length, MIN_METADATA_COLUMN_WIDTH);
+
+        return ({ index }) => {
+            if (index === 0) {
+                return FILEICON_COLUMN_WIDTH;
+            }
+
+            if (index === 1) {
+                return FILENAME_COLUMN_WIDTH;
+            }
+
+            return mdColumnWidth;
+        };
+    }
+
+    getGridCell(columnIndex, rowIndex) {
+        const { currentCollection, metadataColumnsToShow }: Props = this.props;
+        const { items } = currentCollection;
+        const item = items[rowIndex];
+        const { name, metadata } = item;
+
+        if (columnIndex === 0) {
+            return <FileIcon dimension={FILEICON_SIZE} extension={getFileExtension(name) || ''} />;
+        }
+
+        if (columnIndex === 1) {
+            return <span>{name}</span>;
+        }
+
+        const { data } = metadata;
+        const mdFieldName = metadataColumnsToShow[columnIndex - FIXED_COLUMNS_NUMBER];
+        return <span>{data[mdFieldName]}</span>;
+    }
+
+    getGridHeader(columnIndex, style) {
+        const { intl, metadataColumnsToShow } = this.props;
+
+        let displayName;
+
+        switch (columnIndex) {
+            case 0:
+                displayName = null; // No column header for file icon
+                break;
+            case 1:
+                displayName = <span>{intl.formatMessage(messages.name)}</span>; // Column header "Name" for fileName
+                break;
+            default:
+                displayName = <span>{metadataColumnsToShow[columnIndex - FIXED_COLUMNS_NUMBER]}</span>; // Column to show
+        }
+
+        return (
+            <div className="bdl-MetadataBasedItemList-gridHeader" style={style} type="button">
```

Comment:
Should this element be a `Button` instead of a `div`?

Reply:
It's a `<div>`. I will remove the `type` attribute that I left there by mistake.